src/main/java/com/volasoftware/tinder/controllers/AuthenticationController.java

```diff
@@ -22,49 +23,67 @@
+          @ApiResponse(responseCode = "404", description = "Account not found!"),
+          @ApiResponse(responseCode = "406", description = "Email not valid!")
+      }
+  )
+  @PostMapping(value = "/register",
+      consumes = {"application/xml", "application/json"})
+  public ResponseEntity<?> register(
+      @ApiParam(value = "Registration details", required = true)
+      @Valid
+      @RequestBody RegisterRequest registerRequest) {
 
-        return ResponseHandler.generateResponse(
-                AccountConstant.REGISTERED,
-                HttpStatus.OK,
-                authenticationService.register(registerRequest));
-    }
+    return ResponseHandler.generateResponse(
+        AccountConstant.REGISTERED,
+        HttpStatus.OK,
+        authenticationService.register(registerRequest));
+  }
 
-    @ApiOperation(value = "Login Account")
-    @ApiResponses(
-            value = {
-                    @ApiResponse(responseCode = "200", description = "Successfully logged in!"),
-                    @ApiResponse(responseCode = "403", description = "Bad credentials!"),
-                    @ApiResponse(responseCode = "400", description = "Account not verified!"),
-                    @ApiResponse(responseCode = "404", description = "Account not found!")
-            }
-    )
-    @PostMapping(value = "/login",
-            consumes = {"application/xml", "application/json"})
-    public ResponseEntity<?> login(
-            @ApiParam(value = "Login details", required = true)
-            @Valid
-            @RequestBody LoginRequest loginRequest) {
+  @ApiOperation(value = "Login Account")
+  @ApiResponses(
+      value = {
+          @ApiResponse(responseCode = "200", description = "Successfully logged in!"),
+          @ApiResponse(responseCode = "403", description = "Bad credentials!"),
+          @ApiResponse(responseCode = "400", description = "Account not verified!"),
+          @ApiResponse(responseCode = "404", description = "Account not found!")
+      }
+  )
+  @PostMapping(value = "/login",
+      consumes = {"application/xml", "application/json"})
+  public ResponseEntity<?> login(
+      @ApiParam(value = "Login details", required = true)
+      @Valid
+      @RequestBody LoginRequest loginRequest) {
 
-        return ResponseHandler.generateResponse(
-                AccountConstant.LOGGED_IN,
-                HttpStatus.OK,
-                authenticationService.login(loginRequest));
-    }
+    return ResponseHandler.generateResponse(
+        AccountConstant.LOGGED_IN,
+        HttpStatus.OK,
+        authenticationService.login(loginRequest));
+  }
+
+  @ApiOperation(value = "Get Profile of logged Account")
+  @ApiResponses(
+      value = {
+          @ApiResponse(responseCode = "200", description = "Successfully operation!"),
+          @ApiResponse(responseCode = "404", description = "Account not found!")
+      }
+  )
+  @GetMapping(value = "/profile")
+  public ResponseEntity<?> getProfile(
```

Comment:
Do you know what the exact return type of ResponseEntity is? If yes, just use it instead of using <?>. It's just recommendation for more clearity

Reply:
I`m using method "ResponseHandler.generateResponse(String message, HttpStatus status, Object responseObj) for unified response structure and the responseObj in that response can be any type, based on which method is called

Reply:
Can you make it ResponseEntity<{the type of Object response}>?

Reply:
The requirement from TIN-21 is to have unified response structure, thats why the method returns <?>

Reply:
![image](https://github.com/tsenovi/spring-tinder-application/assets/92874246/047c1a97-2870-43b1-b0a7-b971edb4479c)

Reply:
Yes, I know that response is generic, but I'm asking what is the type (dto class) that returns, because it's not visible in the PR

Reply:
It is AccountDto,  returned by AuthenticationService.getAccountByEmail(String email);